src/libraries/Native/Unix/System.Native/pal_memory.c

```diff
@@ -1,27 +1,54 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#include "pal_config.h"
 #include "pal_memory.h"
 
 #include <stdlib.h>
 #include <string.h>
 
-void* SystemNative_MemAlloc(uintptr_t size)
+void* SystemNative_AlignedAlloc(uintptr_t alignment, uintptr_t size)
 {
-    return malloc(size);
+#if HAVE_ALIGNED_ALLOC && !defined(__APPLE__)
+    // We want to prefer the standardized aligned_alloc function. However
+    // it cannot be used on __APPLE__ since we target 10.13 and it was
+    // only added in 10.15, but we might be compiling on a 10.15 box.
+    return aligned_alloc(alignment, size);
+#elif HAVE_POSIX_MEMALIGN
+    void* result = NULL;
+    posix_memalign(&result, alignment, size);
+    return result;
+#else
+    #error "Platform doesn't support aligned_alloc or posix_memalign"
+#endif
+}
+
+void SystemNative_AlignedFree(void* ptr)
```

Comment:
If SystemNative_AlignedFree is the same as SystemNative_Free, do we need both defined in the native code?  Is this an attempt at future proofing, or something else?

Reply:
Its for future proofing. Not all variants of an `AlignedAlloc` API (for example `_aligned_malloc` on Windows) work with `free` and this makes it much more obvious to someone say browsing source that this isn't necessarily "free".